src/client/datascience/kernel-launcher/kernelFinder.ts

```diff
@@ -107,7 +108,16 @@ export class KernelFinder implements IKernelFinder {
         this.writeCache().ignoreErrors();
 
         // ! as the has and set above verify that we have a return here
-        return this.workspaceToKernels.get(workspaceFolderId)!;
+        const promise = this.workspaceToKernels.get(workspaceFolderId)!;
+        promise
+            .then((items) =>
+                traceInfoIf(
```

Comment:
Logging on CI for diagnostics, found that we don't log kernels that we have discovered.